File: evals/refs/mcpi.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	in := bufio.NewScanner(os.Stdin)
	in.Buffer(make([]byte, 1<<20), 1<<20)
	in.Split(bufio.ScanWords)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()
	read := func() string { in.Scan(); return in.Text() }

	const MULT uint64 = 6364136223846793005
	const INC uint64 = 1442695040888963407
	const R2 uint64 = (1 << 30) * (1 << 30)

	T, _ := strconv.Atoi(read())
	for c := 0; c < T; c++ {
		N, _ := strconv.Atoi(read())
		seed, _ := strconv.ParseUint(read(), 10, 64)
		state := seed
		var count uint64 = 0
		for i := 0; i < N; i++ {
			state = state*MULT + INC
			x := (state >> 33) & 0x3FFFFFFF
			state = state*MULT + INC
			y := (state >> 33) & 0x3FFFFFFF
			if x*x+y*y < R2 {
				count++
			}
		}
		fmt.Fprintln(out, count)
	}
}

File: evals/refs/mcpi.c
#include <stdio.h>
#include <stdint.h>

int main(void) {
    int T;
    if (scanf("%d", &T) != 1) return 1;
    const uint64_t MULT = 6364136223846793005ULL;
    const uint64_t INC  = 1442695040888963407ULL;
    const uint64_t R2 = (uint64_t)(1ULL << 30) * (uint64_t)(1ULL << 30);
    while (T--) {
        unsigned long long N, seed;
        if (scanf("%llu %llu", &N, &seed) != 2) return 1;
        uint64_t state = seed;
        uint64_t count = 0;
        for (uint64_t i = 0; i < N; i++) {
            state = state*MULT + INC;
            uint64_t x = (state >> 33) & 0x3FFFFFFF;
            state = state*MULT + INC;
            uint64_t y = (state >> 33) & 0x3FFFFFFF;
            if (x*x + y*y < R2) count++;
        }
        printf("%llu\n", (unsigned long long)count);
    }
    return 0;
}
